backend/server/server.go

```diff
@@ -1036,3 +1040,122 @@ Click "Approve" button to apply the schema update.`,
 
 	return nil
 }
+
+func (s *Server) backfillInstanceChangeHistory(ctx context.Context) {
+	err := func() error {
+		migratedInstanceList, err := s.store.ListInstanceHavingInstanceChangeHistory(ctx)
+		if err != nil {
+			return err
+		}
+		instanceMigrated := make(map[int]bool)
+		for _, id := range migratedInstanceList {
+			instanceMigrated[id] = true
+		}
+
+		instanceList, err := s.store.ListInstancesV2(ctx, &store.FindInstanceMessage{})
+		if err != nil {
+			return err
+		}
+
+		principalList, err := s.store.GetPrincipalList(ctx)
+		if err != nil {
+			return err
+		}
+		nameToPrincipal := make(map[string]*api.Principal)
+		for _, principal := range principalList {
+			nameToPrincipal[principal.Name] = principal
+		}
+
+		var errList error
+		for _, instance := range instanceList {
+			if instance.Engine == db.Redis {
+				continue
+			}
+			if instanceMigrated[instance.UID] {
+				continue
+			}
+
+			databaseList, err := s.store.ListDatabases(ctx, &store.FindDatabaseMessage{InstanceID: &instance.ResourceID, ShowDeleted: true})
+			if err != nil {
+				errList = multierr.Append(errList, err)
+				continue
+			}
+			nameToDatabase := make(map[string]*store.DatabaseMessage)
```

Comment:
We don't need to backfill if there's already instance change history for the  instance.